### ISSUE TITLE
TC not shown if multiple imports from same module

### ISSUE THREAD

**Issue by alanfranz** [snok/flake8-type-checking#136]
Consider this scenario:

```
from collections.abc import Iterator
from collections.abc import Collection


def something() -> Iterator[int]:
    yield 1

class Something(Collection[int]):
    pass
```

We're importing from the very same collections.abc module; if I launch flake8 against this, I get no errors, even though Iterator could be moved in a TYPE_CHECKING block.

My impression (I couldn't verify it thoroughly) is that the module rather than the single "collections.abc.Iterator" import is "whitelisted" because of the actual usage of Collection while subclassing.

**Comment by alanfranz** [comment]
Unit test failure example:

```
diff --git a/tests/test_tc001_to_tc003.py b/tests/test_tc001_to_tc003.py
index d96768a..22f1bd2 100644
--- a/tests/test_tc001_to_tc003.py
+++ b/tests/test_tc001_to_tc003.py
@@ -38,6 +38,7 @@ def get_tc_001_to_003_tests(import_: str, ERROR: str) -> L:
         # Aliased imports
         (f'import {import_} as x\ny:x', {'1:0 ' + ERROR.format(module='x')}),
         (f'from {import_} import constants as x\ny:x', {'1:0 ' + ERROR.format(module='x')}),
+        (f'from {import_} import constants as x\nfrom {import_} import something as q\ny:x\nprint(q)', {'1:0 ' + ERROR.format(module='x')}),
     ]
 
     # These imports are all used. None should generate errors.
```

I just copied the first case, and added an additional (used) "q" import, which is used; this is the result of the test:

```
____________ test_TC001_errors[from flake8_type_checking import constants as x\nfrom flake8_type_checking import something as q\ny:x\nprint(q)-expected7] ____________

example = 'from flake8_type_checking import constants as x\nfrom flake8_type_checking import something as q\ny:x\nprint(q)'
expected = {"1:0 TC001 Move application import 'x' into a type-checking block"}

    @pytest.mark.parametrize(('example', 'expected'), get_tc_001_to_003_tests(mod, TC001))
    def test_TC001_errors(example: str, expected: set[str]) -> None:
>       assert _get_error(example, error_code_filter='TC001') == expected
E       assert set() == {'1:0 TC001 M...ecking block'}
E         Extra items in the right set:
E         "1:0 TC001 Move application import 'x' into a type-checking block"
E         Use -v to get more diff
```

(similar problems for TC002 and TC003)

**Comment by sondrelg** [comment]
Yes you're right, the module is the one we whitelist. This is actually the change that made us bump to v2 not too long ago, since we couldn't think of a good reason for why the plugin should be as pedantic as flagging individual imports.

The plugin provides (in my view) a few benefits. The main ones are 

- Not having to import extra code at runtime, just to annotate your functions
- Avoid import circularity problems

but neither of these actually apply if you've already imported something from a module. More context [here](https://github.com/snok/flake8-type-checking/issues/84) 🙂 

-----------------

That being said, is there another reason why you would prefer the plugin to behave the way you expected? It wouldn't be very hard to add a setting to opt into this behaviour if there was a benefit to it 👍

**Comment by alanfranz** [comment]
ok,
so this intentional. Your rationale is something like "once you have imported a module, it doesn't make much difference if you import one, two, three names in the current namespace", is that right?

It makes sense, I just didn't think about it.

In the next days I'll take a look at the code where I'm planning to use the checker and see if there's any case where I could actually want to separate imports even if they are from the same module.

**Comment by sondrelg** [comment]
Yep that's right. Sounds good 👍 I'll close this for now, and yeah just ping me here if you find a use-case 🙂
